wait_for_deploy.py

```diff
@@ -45,9 +85,40 @@ async def wait_for_deploy(
             ["git", "rev-parse", f"origin/{watch_branch}"], cwd=working_dir
         )
         latest_hash = output.decode().strip()
-    while await fetch_release_hash(hash_url) != latest_hash:
+
+    if expected_version:
+        log.info(
+            "Expecting version '%s' for hash %s", expected_version, latest_hash[:7]
+        )
+    else:
+        log.info(
+            "No specific version expected for hash %s. Proceeding without version check.",
+            latest_hash[:7],
+        )
+
+    while True:
+        try:
+            current_hash = await fetch_release_hash(
+                hash_url, expected_version=expected_version
+            )
```

Comment:
I'm not sure if I follow how this is going to change the current behavior, it still looks like this is going to loop infinitely until the expected hash on the `release` branch shows up.

Reply:
Nevermind, just read it more closely. Addressing...